packages/analytics/dashboard-renderer/src/components/QueryDataProvider.vue

```diff
@@ -133,6 +134,6 @@ const { state, swrvState: STATE } = useSwrvState(v4Data, error, isValidating)
 
 const errorMessage = ref<string | null>(null)
 const hasError = computed(() => state.value === STATE.ERROR || !!errorMessage.value)
-const isLoading = computed(() => !props.queryReady || state.value === STATE.PENDING)
+const isLoading = computed(() => !props.queryReady || state.value === STATE.PENDING || state.value === STATE.VALIDATING_HAS_DATA || state.value === STATE.VALIDATING)
```

Comment:
Minor, but could you make this use the same logic as the one above?  Easier to change that way if we decide want to.